python/GafferSceneUI/_SceneViewInspector.py

```diff
@@ -0,0 +1,959 @@
+			edit = self.__findEdit( p, editScope )
+			if edit is not None :
+				return edit
+
+		if editScope is not None and not self.__errorMessage:
+			# If we've not been able to find an edit, and we have a scope
+			# specified, then we musn't have passed through the edit scope
+			self.__error( "The target Edit Scope '{}' is not in the scene history.", editScope )
+
+		return None
+
+	def __warn( self, message, node ) :
+
+		self.__warningMessages.append( message.format( self.__displayName( node ) ) )
+
+	def __error( self, message, node ) :
+
+		self.__errorMessage = message.format( self.__displayName( node ) )
+		# We clear out the warnings in the case of an error to prevent confusion in the UI
+		self.__warningMessages = []
+
+	@staticmethod
+	def __displayName( node ) :
+
+		return node.relativeName( node.ancestor( Gaffer.ScriptNode ) )
+
+## \todo Figure out how this relates to the DiffRow in the SceneInspector.
+class _ParameterWidget( GafferUI.Widget ) :
+
+	def __init__( self, parameter ) :
+
+		self.__parameter = parameter
+
+		grid = GafferUI.GridContainer( spacing = 2 )
+		GafferUI.Widget.__init__( self, grid )
+
+		self.__inspectors = []
+
+		with grid :
+
+			GafferUI.Label(
+				## \todo Prettify label text (remove snake case)
+				text = "<h5>" + IECore.CamelCase.toSpaced( parameter ) + "</h5>",
+				parenting = { "index" : ( slice( 0, 2 ), 0 ) }
+			)
+
+			self.__editButton = GafferUI.Button( image = "editOff.png", hasFrame = False,
+				parenting = {
+					"index" : ( 0, 1 ),
+					"alignment" : ( GafferUI.HorizontalAlignment.Center, GafferUI.VerticalAlignment.Center )
+				}
+			)
+			self.__editButton.clickedSignal().connect( Gaffer.WeakMethod( self.__editButtonClicked ), scoped = False )
+
+			self.__valueWidget = _ValueWidget( parenting = { "index" : ( 1, 1 ) } )
+			self.__valueWidget.buttonReleaseSignal().connect( Gaffer.WeakMethod( self.__valueWidgetClicked ), scoped = False )
+
+		self.update( [] )
+
+	def update( self, inspectors ) :
+
+		# The somewhat intricate logic here is attempts to determine whether we should present the user
+		# with an edit button for the parameter for the selected locations. This needs to consider:
+		#
+		# - Is the whole selection editable, we deliberately don't allow any edits, if any locations aren't.
+		# - Is there only on type of edit. We can't concurrently edit tweaks and standard numeric node plugs.
+		# - Do all locations currently have edits. We allow editing in this case, but only if we can create the edits.
+		# - In 'Automatic' Edit Scope mode, is an existing edit in a scope or not.
+		#
+		# Based on this, we update the styling of the UI to attempt to clue the user into the current state.
+
+		inspectors = [ i for i in inspectors if i.value() is not None ]
+
+		# Only show something if we have any inspectors for this parameter
+		self.setVisible( inspectors )
+
+		# Display the current values (the widget handles values being different via the multiple value placeholder).
```

Comment:
<del>the widget handles values being different via the multiple value placeholder</del>

The way I'm seeing it, it is entirely up to the ValueWidget how it displays multiple values. In the future refactor, I'm expecting the SceneInspectors DiffWidgets just to be ValueWidgets that choose to show multiple values as diffs.

Reply:
Removed [here](https://github.com/GafferHQ/gaffer/pull/3832/commits/3133a70aab9f52308490a9c4e43de4d968e74e64#diff-b9d160a51cf1199c9bca6867338d8f42L674-R678).